### PR TITLE
Restore default ChildPart.initial_visibility=None

### PR DESCRIPTION
**Pull request by thomascobb**
This means "True if the child Block is connected to anything"

### CODE REVIEW

**Review by BenBradnick — APPROVED** [review]
Looks good, sorry for messing this up trying to make mypy happy!

I see that the layout already has some more detailed tests with parts, so I've just added a couple of tests based on checking the visibility attribute is initialised in the first call to on_layout.

Do you thing any other testing is required? Happy to add more if so.

File: tests/test_modules/test_builtin/test_childpart.py [hunk 1]
```diff
@@ -105,6 +105,7 @@ def test_init(self):
         assert info_out.name == "sourceportConnector"
         assert info_out.port == Port.INT32
         assert info_out.connected_value == "Connector1"
+        assert self.c.block_view().layout.value.visible == [True, True, True]
```
Comment (thomascobb) [inline tests/test_modules/test_builtin/test_childpart.py:108]: This test fails on master and passes here, so this should be sufficient (with the initial_visibility=True removed above)

File: tests/test_modules/test_builtin/test_childpart.py [hunk 2]
```diff
@@ -154,3 +155,37 @@ def test_load_save(self):
         assert structure2 == expected
         self.p1.on_load(context, dict(sinkportConnector="blah_again"))
         assert b1.sinkportConnector.value == "blah_again"
+
+    def test_on_layout_first_call_sets_part_visibility_to_False(self):
```
Comment (thomascobb) [inline tests/test_modules/test_builtin/test_childpart.py:159]: I'm not a big fan of mocked tests, I try to only use them to test the last few corner cases, most of the time I try and cover with integration tests like the rest of this file. I'll leave it up to you whether these add anything or not...

Comment (BenBradnick) [reply tests/test_modules/test_builtin/test_childpart.py]: Sure, I agree that it makes a lot of sense to use the integration test which now covers this check anyway.

**Comment by thomascobb** [comment]
Looks good
